src/earn/EarnCollectScreen.tsx

```diff
@@ -39,38 +39,54 @@ type Props = NativeStackScreenProps<StackParamList, Screens.EarnCollectScreen>
 export default function EarnCollectScreen({ route }: Props) {
   const { t } = useTranslation()
   const dispatch = useDispatch()
-  const { depositTokenId, poolTokenId } = route.params
-  const depositToken = useTokenInfo(depositTokenId)
-  const poolToken = useTokenInfo(poolTokenId)
+  const { pool } = route.params
+  const { depositTokenId } = pool.dataProps
   const withdrawStatus = useSelector(withdrawStatusSelector)
+  const positionsWithClaimableRewards = useSelector(positionsWithClaimableRewardsSelector)
+  const position = useSelector(positionsWithBalanceSelector).find(
+    (position) => position.positionId === pool.positionId
+  )
+  const depositToken = position?.tokens.find((token) => token.tokenId === depositTokenId)
+
+  const rewardsTokens = positionsWithClaimableRewards
+    .filter(
+      (position) =>
+        position.address === pool.address &&
+        position.networkId === pool.networkId &&
+        position.appId === pool.appId
+    )
+    .flatMap((position) => position.claimableShortcut.claimableTokens)
 
-  if (!depositToken || !poolToken) {
+  if (!depositToken || !position) {
     // should never happen
-    throw new Error('Deposit / pool token not found')
+    throw new Error('Deposit token or position not found')
   }
 
   const isGasSubsidized = isGasSubsidizedForNetwork(depositToken.networkId)
 
   const feeCurrencies = useSelector((state) => feeCurrenciesSelector(state, depositToken.networkId))
-  const { asyncRewardsInfo, asyncPreparedTransactions } = useAaveRewardsInfoAndPrepareTransactions({
-    poolTokenId,
+
+  // TODO(ACT-1343): refactor this function using hooks & make this function reusable acroos pools
+  const { asyncPreparedTransactions } = useAaveRewardsInfoAndPrepareTransactions({
+    poolTokenId: pool.tokenId,
     depositTokenId,
     feeCurrencies,
+    rewardsTokens,
   })
   const onPress = () => {
-    if (!asyncRewardsInfo.result || asyncPreparedTransactions.result?.type !== 'possible') {
+    if (asyncPreparedTransactions.result?.type !== 'possible') {
       // should never happen because button is disabled if withdraw is not possible
       throw new Error('Cannot be called without possible prepared transactions')
     }
 
-    const serializedRewards = asyncRewardsInfo.result.map((info) => ({
-      amount: info.amount.toString(),
-      tokenId: info.tokenInfo.tokenId,
+    const serializedRewards = rewardsTokens.map((token) => ({
+      amount: token.balance,
+      tokenId: token.tokenId,
     }))
 
     dispatch(
       withdrawStart({
-        amount: poolToken.balance.toString(),
+        amount: depositToken.balance,
```

Comment:
shouldn't this be the withdrawToken? I assume the depositToken's balance goes down as we deposit

Reply:
oh wow good catch